src/java.base/share/classes/java/lang/foreign/MemorySegment.java

```diff
@@ -608,8 +598,9 @@ default MemorySegment copyFrom(MemorySegment src) {
      * @param elementLayout the source element layout. If the byte order associated with the layout is
      * different from the native order, a byte swap operation will be performed on each array element.
      * @return a fresh short array copy of this memory segment.
```

Comment:
Maybe use "new" instead of "fresh" here and in the other MemorySegment::toArray methods?